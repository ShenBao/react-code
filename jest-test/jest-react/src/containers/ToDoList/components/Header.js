import React, { Component } from 'react';

class Header extends Component {

    constructor(props) {
        super(props);
        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleInputKeyUp = this.handleInputKeyUp.bind(this);

        this.state = {
            value: ''
        };
    }

    handleInputChange(e) {
        this.setState({
            value: e.target.value
        })
    }

    handleInputKeyUp(e) {
        const { value } = this.state;
        if (e.keyCode === 13 && value) {
            this.props.addUndoItem(value);
            this.setState({
                value: ''
            })
        }
    }

    render() {
        const { value } = this.state;
        return <div>
            <h1>ToDoList</h1>
            <input type="text" data-test="input"
                value={value}
                onChange={this.handleInputChange}
                onKeyUp={this.handleInputKeyUp}
                placeholder="ToDoList Input"
            />
        </div>
    }
}

export default Header;
