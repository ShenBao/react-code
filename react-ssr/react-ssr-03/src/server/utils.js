import React from 'react';
import {renderToString} from 'react-dom/server';
import {StaticRouter} from 'react-router-dom';

import Routers from '../Routers';

export const renderPage = (req)=> {
    const App = () => {
        return (
          <StaticRouter location={req.path} context={{}}>
            {Routers}
          </StaticRouter>
        );
      };
    
      const content = renderToString (<App />);
    
      return `
        <!DOCTYPE html>
        <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1.0">
            <title>React SSR</title>
        </head>
        <body>
            <div id="root">${content}</div>
            <script src='/index.js'></script>
        </body>
        </html>
        `
      ;
}