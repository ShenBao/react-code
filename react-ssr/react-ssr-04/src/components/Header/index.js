import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => {
    return (
      <div>
        <Link to='/'>Home</Link>
        {'  '}
        <Link to='/login'>Login</Link>
        <hr />
      </div>
    );
  };
  
  export default Header;

