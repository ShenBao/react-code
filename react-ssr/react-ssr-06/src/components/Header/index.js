import React, {Component, Fragment} from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {actions} from './store/';

class Header extends Component {
  render () {
    const {login, handleLogout, handleLogin} = this.props;

    return (
      <div>
        <Link to="/">Home</Link>
        {'  '}
        {login
          ? <Fragment>
              <Link to="/list">List</Link>
              {'  '}
              <span onClick={handleLogout}>Logout</span>
            </Fragment>
          : <Fragment>
              <span onClick={handleLogin}>Login</span>
            </Fragment>}
        <hr />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  login: state.header.login,
});

const mapDispatchToProps = dispatch => ({
  handleLogin () {
    dispatch (actions.login ());
  },
  handleLogout () {
    dispatch (actions.logout ());
  },
});

export default connect (mapStateToProps, mapDispatchToProps) (Header);
