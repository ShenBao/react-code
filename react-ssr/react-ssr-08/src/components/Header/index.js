import React, {Component, Fragment} from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {actions} from './store/';
import style from './style.css';
import withStyle from '../../withStyle';

class Header extends Component {
  render () {
    const {login, handleLogout, handleLogin} = this.props;

    return (
      <div className={style.header}>
        <Link to="/">Home</Link>
        {login
          ? <Fragment>
              <Link to="/list">List</Link>
              <span onClick={handleLogout}>Logout</span>
            </Fragment>
          : <Fragment>
              <span onClick={handleLogin}>Login</span>
            </Fragment>}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  login: state.header.login,
});

const mapDispatchToProps = dispatch => ({
  handleLogin () {
    dispatch (actions.login ());
  },
  handleLogout () {
    dispatch (actions.logout ());
  },
});

export default connect (mapStateToProps, mapDispatchToProps) (
  withStyle (Header, style)
);
