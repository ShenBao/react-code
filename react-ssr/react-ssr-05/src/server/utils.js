import React from 'react';
import {renderToString} from 'react-dom/server';
import {StaticRouter, Route} from 'react-router-dom';

import {Provider} from 'react-redux';

export const renderPage = (store, routes, req) => {
  const App = () => {
    return (
      <Provider store={store}>
        <StaticRouter location={req.path} context={{}}>
          <div>
            {routes.map (route => {
              return <Route key={route.key} {...route} />;
            })}
          </div>
        </StaticRouter>
      </Provider>
    );
  };

  const content = renderToString (<App />);
  const stateData = JSON.stringify (store.getState ());

  return `
    <!DOCTYPE html>
    <html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>React SSR</title>
    </head>
    <body>
        <div id="root">${content}</div>
        <script>
          window.context = {
            state: ${stateData}
          };
        </script>
        <script src='/index.js'></script>
    </body>
    </html>
  `;
};
