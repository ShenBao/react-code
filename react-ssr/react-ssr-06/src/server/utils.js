import React from 'react';
import {renderToString} from 'react-dom/server';
import {StaticRouter, Route} from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import {Provider} from 'react-redux';

export const renderPage = (store, routes, req, context) => {
  const App = () => {
    return (
      <Provider store={store}>
        <StaticRouter location={req.path} context={context}>
          <div>
            {renderRoutes(routes)}
          </div>
        </StaticRouter>
      </Provider>
    );
  };

  const content = renderToString (<App />);
  const stateData = JSON.stringify (store.getState ());

  return `
    <!DOCTYPE html>
    <html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>React SSR</title>
    </head>
    <body>
        <div id="root">${content}</div>
        <script>
          window.context = {
            state: ${stateData}
          };
        </script>
        <script src='/index.js'></script>
    </body>
    </html>
  `;
};
