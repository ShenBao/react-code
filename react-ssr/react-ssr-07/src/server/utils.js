import React from 'react';
import {renderToString} from 'react-dom/server';
import {StaticRouter, Route} from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import {Provider} from 'react-redux';

export const renderPage = (store, routes, req, context) => {
  const App = () => {
    return (
      <Provider store={store}>
        <StaticRouter location={req.path} context={context}>
          <div>
            {renderRoutes(routes)}
          </div>
        </StaticRouter>
      </Provider>
    );
  };

  const content = renderToString (<App />);
  const styleContent= context.css.length ? context.css.join('\n') :'';
  const stateData = JSON.stringify (store.getState ());

  return `
    <!DOCTYPE html>
    <html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>React SSR</title>
        <style>${styleContent}</style>
    </head>
    <body>
        <div id="root">${content}</div>
        <script>
          window.context = {
            state: ${stateData}
          };
        </script>
        <script src='/index.js'></script>
    </body>
    </html>
  `;
};
