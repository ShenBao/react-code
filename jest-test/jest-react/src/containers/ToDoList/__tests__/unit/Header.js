import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

import { findTestWrapper } from './../../../../utils/testUtils';

describe('Header 组件', () => {
    it('渲染正常', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });
    it('组件包含一个 input 框', () => {
        const wrapper = shallow(<Header />);
        // const inputElem = wrapper.find('[data-test="input"]');
        const inputElem = findTestWrapper(wrapper, 'input');
        expect(inputElem.length).toBe(1);
    });

    it('组件 input 框, 初始化应该为空', () => {
        const wrapper = shallow(<Header />);
        // const inputElem = wrapper.find('[data-test="input"]');
        const inputElem = findTestWrapper(wrapper, 'input');
        expect(inputElem.prop('value')).toEqual('');
    });

    it('input 框, 用户输入时，会跟随变化', () => {
        const wrapper = shallow(<Header />);
        // const inputElem = wrapper.find('[data-test="input"]');
        const inputElem = findTestWrapper(wrapper, 'input');
        const uesrInput = 'Jest test input';
        inputElem.simulate('change', {
            target: {
                value: uesrInput
            }
        })
        expect(wrapper.state('value')).toEqual(uesrInput);
        // const newInputElem = wrapper.find('[data-test="input"]');
        // expect(newInputElem.prop('value')).toEqual(uesrInput);
    });
    it('input 框按回车，如果 input 无内容，无操作', () => {
        const fn = jest.fn();
        const wrapper = shallow(<Header addUndoItem={fn} />);
        // const inputElem = wrapper.find('[data-test="input"]');
        const inputElem = findTestWrapper(wrapper, 'input');
        wrapper.setState({
            value: ''
        });
        inputElem.simulate('keyUp', {
            keyCode: 13
        });
        expect(fn).not.toHaveBeenCalled();
    });

    it('input 框按回车是，如果 input 有内容，函数被调用有操作', () => {
        const fn = jest.fn();
        const wrapper = shallow(<Header addUndoItem={fn} />);
        // const inputElem = wrapper.find('[data-test="input"]');
        const inputElem = findTestWrapper(wrapper, 'input');
        wrapper.setState({
            value: '学习 React Jest'
        });
        inputElem.simulate('keyUp', {
            keyCode: 13
        });
        expect(fn).toHaveBeenCalled();
        expect(fn).toHaveBeenCalledWith('学习 React Jest');
    });

    it('input 框按回车，如果 input 有内容，应该被清空', () => {
        const fn = jest.fn();
        const wrapper = shallow(<Header addUndoItem={fn} />);
        // const inputElem = wrapper.find('[data-test="input"]');
        const inputElem = findTestWrapper(wrapper, 'input');
        wrapper.setState({
            value: '学习 React Jest'
        });
        inputElem.simulate('keyUp', {
            keyCode: 13
        });
        // const newInputElem = wrapper.find('[data-test="input"]');
        const newInputElem = findTestWrapper(wrapper, 'input');
        expect(newInputElem.prop('value')).toBe('');
    });
});
