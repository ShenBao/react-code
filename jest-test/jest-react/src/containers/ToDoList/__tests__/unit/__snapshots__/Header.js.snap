// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 组件 渲染正常 1`] = `
<div>
  <h1>
    ToDoList
  </h1>
  <input
    data-test="input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="ToDoList Input"
    type="text"
    value=""
  />
</div>
`;

exports[`Header 组件包含一个 input 框 1`] = `
<div>
  <h1>
    ToDoList
  </h1>
  <input
    data-test="input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="ToDoList Input"
    type="text"
    value=""
  />
</div>
`;
